EIPS/eip-5169.md

```diff
@@ -0,0 +1,191 @@
+
+It's important that a token's client script is linked with the token's contract, since the client script carry out trusted tasks such as creating transactions for the user.
+
+Users can be sure they are using the correct script through the contract author packaging a URI to an official script, made available with a call to the token contract itself.
+
+This ERC proposes adding a `scriptURI`, a structure containing URIs to download the client script. It can be a link to IPFS, GitHub, a cloud provider, etc.
+
+Each token contract has one `scriptURI`  function to return the download URI to a client script.
+
+Concretely each element in the array contains a URI to download the script itself.
+
+The script provides a client-side executable to the hosting token. Examples of such a script could be:
+
+- A 'miniDapp', which is a cut-down dapp tailored for a single token.
+- A 'TokenScript' which provides [T.I.P.S.](https://tokenscript.org/TIPS.html) from a browser wallet.
+- An extension that is downloadable to the hardware wallet with an extension framework, such as Ledger.
+
+#### Script location
+
+While the most straightforward solution to facilitate specific script usage associated with NFTs, is clearly to store such a script on the smart contract. However, this has several disadvantages: 
+
+1. The smart contract signing key is needed to make updates, causing the key to become more exposed, as it is used more often. 
+
+2. Updates require smart contract interaction. If frequent updates are needed, smart contract calls can become an expensive hurdle.
+
+3. Storage fee. If the script is large, updates to the script will be costly. A client script is typically much larger than a smart contract.
+
+For these reasons, storing volatile data, such as token enhancing functionality, on an external resource makes sense. Such an external resource can be either be  hosted centrally, such as through a cloud provider, or privately hosted through a private server, or decentralized hosted, such as the interplanetary filesystem.
+
+While centralized storage for a decentralized functionality goes against the ethos of web3, fully decentralized solutions may come with speed, price or space penalties. This ERC handles this by allowing the function `ScriptURI` to return multiple URIs, which could be a mix of centralized, individually hosted and decentralized locations.
+
+While this ERC does not dictate the format of the stored script, the script itself could contain pointers to multiple other scripts and data sources, allowing for advanced ways to expand token scripts, such as lazy loading. 
+The handling of the integrity of such secondary data sources is left dependent on the format of the script. For example, HTML format uses [the `integrity` property](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity), while [signed XML format has `<Manifest/>`](https://www.w3.org/TR/xmldsig-core2/#sec-Manifest).
+
+#### Overview
```

Comment:
This should be moved into a sub-section of the specification.

Reply:
May also be appropriate for the Abstract.  The abstract is meant to be a human readable and pithy description of the standard.  I'll leave it to others to decide if this counts or not.

Reply:
Have moved the Script Location subsection